docs/understanding-airbyte/schemaless-sources-and-destinations.md

```diff
@@ -0,0 +1,53 @@
+# "Schemaless" Sources and Destinations
+
+In order to run a sync, Airbyte requires a [catalog](/understanding-airbyte/airbyte-protocol#catalog), which includes a data schema describing the shape of data being emitted by the source.
+This schema will be used to prepare the destination to populate the data during the sync.
+
+While having a [strongly-typed](/understanding-airbyte/supported-data-types) catalog/schema is possible for most sources, some won't have a reasonably static schema.  This document describes the options available for the subset of sources that do not have a strict schema, aka "schemaless sources".
+
+## What is a Schemaless Source?
+
+Schemaless sources are sources for which there is no requirement or expectation that records will conform to a particular pattern.
+For example, in a MongoDB database, there's no requirement that the fields in one record are the same as the fields in the next, or that the type of value in one field is the same as the type for that field in a separate record.
+Similarly, for a file-based source such as S3, the files that are present in your source may not all have the same schema.
+
+Although the sources themselves may not conform to an obvious schema, Airbyte still needs to know the shape of the data in order to prepare the destination for the records.
+For these sources, during the [`discover`](/understanding-airbyte/airbyte-protocol#discover) method, Airbyte offers two options to create the schema:
+
+1. Dynamic schema inference.
+2. A hardcoded "schemaless" schema.
+
+### Dynamic schema inference
+If this option is selected, Airbyte will infer the schema dynamically based on the contents of the source.  If your source's content is homogenous, we recommend this option, as the data in your destination will be typed and you can make use of schema evolution features, column selection, and similar Airbyte features which operate against the source's schema. 
+For MongoDB, if the dataset exceeds 10,000 documents, the schema is inferred from a random sample of up to 10,000 documents; otherwise, it is inferred from the entire dataset.
```

Comment:
Sample size is configurable in source-mongodb: 1,000 - 100,000

Reply:
👍 updated.